src/service/crypto-impl.js

```diff
@@ -129,102 +123,70 @@ export class Crypto {
   }
 
   /**
-   * Convert a JSON Web Key object to a browser-native cryptographic key and
-   * compute a hash for it.  The caller must verify that Web Cryptography is
-   * available using isCryptoAvailable before calling this function.
+   * Checks whether Web Cryptography is available, which is required for PKCS 1
+   * operations. SHA-384 operations do not need this because there's a polyfill.
+   * This could be false if the browser does not support Web Cryptography, or if
+   * the current browsing context is not secure (e.g., it's on an insecure HTTP
+   * page, or an HTTPS iframe embedded in an insecure HTTP page).
    *
-   * @param {string} serviceName used to identify the signing service.
-   * @param {!JsonObject} jwk An object which is hopefully an RSA JSON Web Key.  The
-   *     caller should verify that it is an object before calling this function.
-   * @return {!Promise<!PublicKeyInfoDef>}
+   * @return {boolean} whether Web Cryptography is available
    */
-  importPublicKey(serviceName, jwk) {
-    dev().assert(this.isCryptoAvailable());
-    // WebKit wants this as an ArrayBufferView.
-    return (this.isWebkit_ ?
-          utf8Encode(JSON.stringify(jwk)) : Promise.resolve(jwk))
-        .then(encodedJwk => this.subtle_.importKey(
-            'jwk',
-            encodedJwk,
-            {name: 'RSASSA-PKCS1-v1_5', hash: {name: 'SHA-256'}},
-            true,
-            ['verify']))
-        .then(cryptoKey => {
-          // We do the importKey first to allow the browser to check for
-          // an invalid key.  This last check is in case the key is valid
-          // but a different kind.
-          if (typeof jwk['n'] != 'string' || typeof jwk['e'] != 'string') {
-            throw new Error('missing fields in JSON Web Key');
-          }
-          const mod = base64UrlDecodeToBytes(jwk['n']);
-          const pubExp = base64UrlDecodeToBytes(jwk['e']);
-          const lenMod = lenPrefix(mod);
-          const lenPubExp = lenPrefix(pubExp);
-          const data = new Uint8Array(lenMod.length + lenPubExp.length);
-          data.set(lenMod);
-          data.set(lenPubExp, lenMod.length);
-          // The list of RSA public keys are not under attacker's
-          // control, so a collision would not help.
-          return this.subtle_.digest({name: 'SHA-1'}, data)
-              .then(digest => ({
-                serviceName,
-                cryptoKey,
-                // Hash is the first 4 bytes of the SHA-1 digest.
-                hash: new Uint8Array(/** @type {ArrayBuffer} */(digest), 0, 4),
-              }));
-        });
+  isPkcsAvailable() {
+    return Boolean(this.subtle_) && this.win_['isSecureContext'] !== false;
```

Comment:
Nit: Not necessary to cast to boolean here. If you do need to cast, `!!value` is more common.